app/controllers/admin/users_controller.rb

```diff
@@ -1,5 +1,17 @@
 module Admin
   class UsersController < ApplicationController
+    def edit
```

Comment:
this is great, but it doesn't check to see if the user doing the promotion is an admin! This would be a big security problem :-)